dbux-code/src/search/searchController.js

```diff
@@ -0,0 +1,87 @@
+import EmptyArray from '@dbux/common/src/util/EmptyArray';
+import SearchMode from '@dbux/graph-common/src/shared/SearchMode';
+import NanoEvents from 'nanoevents';
+import { SearchContext, SearchTrace, SearchValue } from './SearchHelper';
+
+class SearchController {
+  constructor() {
+    this._emitter = new NanoEvents();
```

Comment:
Since this is a singleton, have you made sure that it gets reset when it is invalidated?
Specifically: this stores `matches`, `contexts` etc. When their application is deselected/removed, are you making sure that those invalid data points are not lingering?
Can you add at least part of the reset mechanic as a method to this class, so we can easily find where it happens, when necessary?

Reply:
Solution: `re-search` every time `ApplicationSelectionChanged`.